portfolio/src/main/webapp/script.js

```diff
@@ -12,6 +12,11 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//Global variable for later use of markers
+var map;
+var markerList = [];
+var markersVisible = false;
```

Comment:
I don't think you need this global state.  The markers will keep track if they're visible or not